pkger/parser.go

```diff
@@ -1555,29 +1576,52 @@ func (p *Template) parseQuery(prefix, source string, params []Resource) (query,
 		Query: strings.TrimSpace(source),
 	}
 
+	mParams := make(map[string]*references)
+	tParams := make(map[string]*references)
+
 	opt, err := edit.GetOption(files[0], "params")
-	if err != nil {
-		return q, nil
-	}
-	obj, ok := opt.(*ast.ObjectExpression)
-	if !ok {
+	topt, err2 := edit.GetOption(files[0], "task") // parse the task out of the query. need to use these params to set the task bits
+	if err != nil && err2 != nil {
 		return q, nil
 	}
 
-	mParams := make(map[string]*references)
-	for _, p := range obj.Properties {
-		sl, ok := p.Key.(*ast.Identifier)
-		if !ok {
-			continue
+	if err == nil {
```

Comment:
i'm having a hard time following the flow and logic of this. can we pair up to talk about it tomorrow?